src/partition/repart.c

```diff
@@ -146,6 +146,7 @@ static bool arg_legend = true;
 static void *arg_key = NULL;
 static size_t arg_key_size = 0;
 static EVP_PKEY *arg_private_key = NULL;
+static char *arg_private_key_source = NULL;
```

Comment:
i find this a bit sloppy, we usually want to parse things to the level we understand them before we store them in internal variables. i.e. this really should be two fields: an enum and a char*.

```c
typedef enum KeySourceType {
        KEY_SOURCE_FILE,
        KEY_SOURCE_ENGINE,
        KEY_SOURCE_PROVIDER,
        _KEY_SOURCE_MAX,
        _KEY_SOURCE_INVALID = -EINVAL,
} KeySouceType;

static KeySourceType arg_private_key_source_type = KEY_SOURCE_FILE;
static const char *arg_private_key_source_parameter = NULL;
```
and so on.